File: javatest/src/main/java/com/anymore/javatest/di/ViewModelKey.java
package com.anymore.javatest.di;

import android.arch.lifecycle.ViewModel;
import dagger.MapKey;
import dagger.internal.Beta;

import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.METHOD;

/**
 * Created by liuyuanmao on 2019/2/22.
 */
@Beta
@MapKey
@Target(METHOD)
public @interface ViewModelKey {
    Class<? extends ViewModel> value();
}

File: mvvmkit/src/main/java/com/anymore/mvvmkit/di/key/ViewModelKey.kt
package com.anymore.mvvmkit.di.key

import android.arch.lifecycle.ViewModel
import dagger.MapKey
import dagger.internal.Beta
import kotlin.reflect.KClass

/**
 * Created by liuyuanmao on 2019/2/22.
 */
@Beta
@MapKey
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class ViewModelKey(val value: KClass<out ViewModel>)
